server/server.js

```diff
@@ -13,76 +13,163 @@
 	// #region Imports
 	// NPM
 	const chalk = require("chalk");
+	const errorColor = chalk.red;
+	const outputColor = chalk.white;
 	chalk.enabled = true;
 	//force color output
 	chalk.level = 1;
 	const express = require("express");
 	const fs = require("fs");
 	const path = require("path");
 	const compression = require("compression");
+	const httpProxy = require("http-proxy-middleware");
 	// Local
-	const extensions = fs.existsSync(path.join(__dirname, "server-extensions.js")) ?
-		require("./server-extensions") :
-		{
-			/**
-			 * Method called before starting the server.
-			 *
-			 * @param {function} done Function can take one argument; an error message if one occurred.
-			 			 * @example
-			 * const pre => {
-			 * 	try {
-			 * 		// do something that could throw an error
-			 *  } catch(e) {
-			 *		done(e);
-			 *  }
-			 * }*/
-			pre: done => { done(); },
+
+	const handleError = e => {
+		if (e) {
+			console.error(e);
+			process.send("serverFailed");
+			process.exit(1);
+		}
+	}
+
+	const defaultExtensions = {
+
+		/**
+		 * Method called before starting the server.
+		 *
+		 * @param {function} done Function can take one argument; an error message if one occurred.
+					 * @example
+		 * const pre => {
+		 * 	try {
+		 * 		// do something that could throw an error
+		 *  } catch(e) {
+		 *		done(e);
+			*  }
+			* }*/
+		pre: done => { done(); },
+
+		/**
+		 * Method called after the server has started.
+		 *
+		 * @param {function} done Function can take one argument; an error message if one occurred.
+		 *
+		 * @example
+		 * const post => {
+		 * 	try {
+		 * 		// do something that could throw an error
+		 *  } catch(e) {
+		 *		done(e);
+			*  }
+			* }
+			*/
+		post: done => { done(); },
+
+		/**
+		 * Method called to update the server.
+		 *
+		 * @param {express} app The express server.
+		 * @param {function} cb The function to call once finished adding functionality to the server.
+		 * @example
+		 * const cb => {
+		 * 	try {
+		 * 		// do something that could throw an error
+		 *  } catch(e) {
+		 *		done(e);
+			*  }
+			* }*/
+		updateServer: (app, cb) => {
+			const server = app.listen(
+				PORT,
+				e => {
+					handleError(e);
+
+					logToTerminal(outputColor(`Listening on port ${PORT} `));
+
+					global.host = server.address().address;
+					global.port = server.address().port;
+				});
+
+			app.use(compression());
+			// Sample server root set to "/" -- must align with paths throughout
+			app.use("/", express.static(rootDir, options));
+			// Open up the Finsemble Components,services, and clients
+			app.use("/Finsemble", express.static(moduleDirectory, options));
+			// For Assimilation
+			app.use("/hosted", express.static(path.join(__dirname, "..", "hosted"), options));
+
+			// configs/openfin/manifest-local.json and configs/other/server-environment-startup.json
+			// Make the config public
+			app.use("/configs", express.static("./configs", options));
```

Comment:
This seems inconsistent. The configs folder is copied to dist, but is hosted from the project root. We should either not copy it, or host it from dist. I'm for the hosting it from dist.